### PR TITLE
Allow user to set auto-refresh interval

### PR DESCRIPTION
**Pull request by jaredscheib**
- [x] CHANGELOG.md updated
  - [x] Rebased/mergable
  - [ ] Tests pass

Connect #781 

### The problem
Auto-refresh for graphs was set to a constant interval of 15 seconds. User requested the ability to change this interval.

### The Solution
Created an appConfig reducer and respective action creator to set and persist app-wide autoRefresh interval. Also created a (likely temporary) dropdown menu specifically for AutoRefresh. This first iteration allows for setting a global auto-refresh interval from any page that displays graphs.

screen shot 2017-03-01 at 9 29 56 pm

### CODE REVIEW

**Comment by jaredscheib** [comment]
Tests are pending...

**Comment by goller** [comment]
Hey @jaredscheib, looks great, but, would you be able to turn off the auto-refresh as well? @rossmcdonald asked for this in #279

**Comment by jaredscheib** [comment]
Thanks for pointing me to that, @goller – good idea. So maybe a first or last menu option that says "Never" ?

**Comment by goller** [comment]
@jaredscheib, perhaps `Paused`?  Also, maybe the icon changes from the circular arrows to a pause ?

File: ui/src/shared/components/AutoRefresh.js [hunk 1]
```diff
@@ -6,25 +6,34 @@ function _fetchTimeSeries(source, db, rp, query) {
...
+const {
+  element,
+  number,
+  arrayOf,
+  shape,
+  oneOfType,
+  string,
+} = PropTypes
+
 export default function AutoRefresh(ComposedComponent) {
```
Comment (121watts) [inline ui/src/shared/components/AutoRefresh.js]: Be the change.

File: ui/src/shared/reducers/appConfig.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import {AUTOREFRESH_DEFAULT} from 'src/shared/constants'
+
+const initialState = {
+  autoRefresh: AUTOREFRESH_DEFAULT,
+}
+
+const appConfig = (state = initialState, action) => {
```
Comment (121watts) [inline ui/src/shared/reducers/appConfig.js]: I would either move the stuff that's in appUI into appConfig or visa versa.  I created the appUI to hold global app state which is exactly what this is.  I don't think we need two separate reducers.

**Comment by jaredscheib** [comment]
@alexpaxton Thoughts on Paused icon? And if you're cool with it, could you add one for us?

File: ui/src/shared/reducers/appConfig.js (hunk 2 above)
Comment (jaredscheib) [inline ui/src/shared/reducers/appConfig.js]: @cryptoquick and I discussed this a bit. My opinion was that appUI is for UI-only state (presentation mode), whereas appConfig would be data persisted about user choices for configuring the app. UI and config feel separate to me. Thoughts?

File: ui/src/shared/components/AutoRefresh.js (hunk 1 above)
Comment (jaredscheib) [inline ui/src/shared/components/AutoRefresh.js]: aka missing PropTypes ?

**Comment by alexpaxton** [comment]
@jaredscheib incoming!

File: ui/src/shared/components/AutoRefresh.js (hunk 1 above)
Comment (jaredscheib) [inline ui/src/shared/components/AutoRefresh.js]: actually, there are proptypes, it's just within the React.createClass

**Comment by jaredscheib** [comment]
Looks great - nice work, @cryptoquick !

**Comment by jaredscheib** [comment]
Here's the autoRefresh dropdown with Paused and a paused icon.

@alexpaxton @goller @nhaugo Even though the app will always start out with some >0 autoRefresh interval, I feel conflicted about an AutoRefresh dropdown that shows the above, as it no longer indicates that the dropdown is for AutoRefresh. I can go either way -- thoughts?

screen shot 2017-03-03 at 1 47 26 pm

**Comment by jamesyu0725** [comment]
Is there a way to set the Auto-Refresh in presentation mode? Or force AutoRefresh to stay on ("Not paused"). For some reason, we have seen the dashboard automatically "Pause" while in presentation mode. The user can not see the top menu, so doesn't realize that and complains that the data is not loading. Thoughts?

**Pull request merged**